react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({ IMG: 0 });
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(dir) {
+    if (dir === 'left') {
+      if (this.state.IMG === 0) {
+        this.setState({
+          IMG: 4
+        });
+      } else {
+        this.setState(state => ({
+          IMG: state.IMG - 1
+        }));
+      }
+    } else if (dir === 'right') {
+      if (this.state.IMG === 4) {
+        this.setState({
+          IMG: 0
+        });
+      } else {
+        this.setState(state => ({
+          IMG: state.IMG + 1
+        }));
+      }
+    } else {
+      this.setState({
+        IMG: dir
+      });
+    }
+    clearInterval(this.timerID);
+    this.componentDidMount();
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(
+      () => this.tick(),
```

Comment:
Instead of recreating the same logic in a separate method `tick` you could just use your `handleClick` method here:

```JAVASCRIPT
componentDidMount() {
    this.timerID = setInterval(
      () => this.handleClick('right'),
      3000
    );
 }
 ```